### ISSUE TITLE
[bug] django.test.TestCase.setting() does not work as expected

### ISSUE THREAD

**Issue by dkliban** [dynaconf/dynaconf#689]
I have a unit test in pulpcore that uses TestCase.setting() to override a setting. However, the setting never gets set to the right value. 

```

class ContentHeartbeatTestCase(TestCase):
    @patch("pulpcore.app.models.ContentAppStatus.objects.get_or_create")
    @patch("pulpcore.content.handler.Handler._reset_db_connection")
    def test_db_connection_interface_error(self, mock_reset_db, mock_get_or_create):
        """
        Test that if an InterfaceError or OperationalError is raised,
        Handler._reset_db_connection() is called
        """

        class MockException(Exception):
            pass

        mock_get_or_create.side_effect = [InterfaceError(), OperationalError(), MockException()]

        loop = asyncio.get_event_loop()
        with self.settings(CONTENT_APP_TTL=1):
            try:
                loop.run_until_complete(_heartbeat())
            except MockException:
                pass
        loop.close()

        mock_get_or_create.assert_called()
        mock_reset_db.assert_has_calls([call(), call()])

```

The CONTENT_APP_TTL is always set to the default of 30.

**Comment by rochacbruno** [comment]
@dkliban I have been trying this and used latest django available 3.2.9

on settings.py
```py
CONTENT_APP_TTL = 30
```

On my tests.py

```py

from django.conf import settings
from django.test import TestCase

class TestIssue689(TestCase):
    def test_settings_override_689(self):
        # before
        self.assertEqual(settings.CONTENT_APP_TTL, 30)

        # after
        with self.settings(CONTENT_APP_TTL=1):
            self.assertEqual(settings.CONTENT_APP_TTL, 1)

        # same as before
        self.assertEqual(settings.CONTENT_APP_TTL, 30)
```

REsult

```bash
❯ ./manage.py test -v=2 -k TestIssue689
Creating test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...
Operations to perform:
  Synchronize unmigrated apps: debug_toolbar, messages, staticfiles
  Apply all migrations: admin, auth, contenttypes, sessions
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying sessions.0001_initial... OK
System check identified no issues (0 silenced).
test_settings_override_689 (polls.tests.TestIssue689) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.009s

OK
Destroying test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...

```

So looks like the issue you have might be related to your assertion being run on an async loop, looks like even the general django override settings feature (without dynaconf) doesn't apply changes on async code.

**Comment by rochacbruno** [comment]
this is probably a bug only when code is async.. not sure how to solve..
